lib/util.js

```diff
@@ -20,6 +20,30 @@ export const w3cDate = date => {
   return str.slice(0, - 5) + 'Z';
 };
 
+export const timezoneOffset = new RegExp(
+  '(Z|(\\+|-)((0[0-9]|1[0-3]):[0-5][0-9]|14:00))$');
+
+// Z and T must be uppercase
+// xml schema date time RegExp
+// @see https://www.w3.org/TR/xmlschema11-2/#dateTime
+export const XMLDateTimeRegExp = new RegExp(
+  '-?([1-9][0-9]{3,}|0[0-9]{3})' +
+  '-(0[1-9]|1[0-2])' +
+  '-(0[1-9]|[12][0-9]|3[01])' +
+  'T(([01][0-9]|2[0-3]):[0-5][0-9]:[0-5][0-9](\.[0-9]+)?|(24:00:00(\.0+)?))' +
+  '(Z|(\\+|-)((0[0-9]|1[0-3]):[0-5][0-9]|14:00))?');
+export const isW3cDate = timeStamp => XMLDateTimeRegExp.test(timeStamp);
+
+export const convertTimeStamp = timestamp => {
+  if(!timestamp) {
+    throw new Error(`Expected timestamp recieved: ${timestamp}`);
```

Comment:
Should this say unexpected? I prefer to put the value in quotes in case it just looks "blank" at the end of the colon. Also, spelling fix:

```suggestion
    throw new Error(`Unexpected timestamp ("${timestamp}") received.`);
```